tests/benchmarks/thread_metric/src/tm_porting_layer_zephyr.c

```diff
@@ -69,6 +69,11 @@ int tm_thread_create(int thread_id, int priority, void (*entry_function)(void *,
 			      TM_TEST_STACK_SIZE, entry_function,
 			      NULL, NULL, NULL, priority, 0, K_FOREVER);
 
+	/* Thread started in sleeping state. Switch to suspended state */
+
+	k_thread_suspend(&test_thread[thread_id]);
+	k_wakeup(&test_thread[thread_id]);
```

Comment:
do not get this sequence here, it is sleeping, we suspend it, then we wake it up?

Reply:
The point is to mark the thread "suspended" (which this test requires, and which brand new threads not longer qualify as) without letting it actually run, I suspect.

Reply:
The thread_metric benchmarks assume that the thread was started in the suspended state. Transitioning from a sleep to suspended state allows us to maintain that model for the benchmark and keep tm_thread_resume() simple (which in turns helps with the performance numbers).

Reply:
Hi @peter-mitsis 

> The thread_metric benchmarks assume that the thread was started in the suspended state. Transitioning from a sleep to suspended state allows us to maintain that model for the benchmark and keep tm_thread_resume() simple (which in turns helps with the performance numbers).

Can we add it as an comment to the sources here? So it'll be easier to understand this tricks later if someone have to look at this test.

Thanks!